File: gems/decomplex/examples/ruby/path-condition.rb
# frozen_string_literal: true

def one(x, y, z)
  go(x) if x.p? && y.q? && z.r?
end

def two(x, y, z)
  go(x) if x.p? && y.q? && z.r?
end

def three(x, y, z)
  go(x) if x.p? && y.q? && z.r?
end

def bug(x, y, z)
  go(x) if x.p? && y.q?
end

File: gems/decomplex/examples/rust/path-condition.rs
fn one(x: X, y: Y, z: Z) {
    if x.p() && y.q() && z.r() { go(x); }
}

fn two(x: X, y: Y, z: Z) {
    if x.p() && y.q() && z.r() { go(x); }
}

fn three(x: X, y: Y, z: Z) {
    if x.p() && y.q() && z.r() { go(x); }
}

fn bug(x: X, y: Y, z: Z) {
    if x.p() && y.q() { go(x); }
}
